### PR TITLE
Linq follow up changes on FeedIterator

### PR DESCRIPTION
**Pull request by simplynaveen20**
This is follow up changes of https://github.com/Azure/azure-cosmos-dotnet-v3/pull/461

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/CosmosExtensions.cs [hunk 1]
```diff
@@ -52,7 +53,11 @@ internal static string ToSqlQueryText<T>(this IQueryable<T> query)
         /// </example>
         public static FeedIterator<T> ToFeedIterator<T>(this IQueryable<T> query)
         {
-            return ((CosmosLinqQuery<T>)query).ToFeedIterator();
+            CosmosLinqQuery<T> linqQuery = query as CosmosLinqQuery<T>;
+
+            return linqQuery != null
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/CosmosExtensions.cs]: Please change to simple check and throw. Easier to read. 
Typically ternary used when result is same, here not.

Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/CosmosExtensions.cs]: With C# 7 this is a feature added to throw between the expression , isn't it ? 
https://structuredsight.com/2016/09/01/c-7-additions-throw-expressions/
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#throw-expressions

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/CosmosExtensions.cs]: Logically its correct. But readability is not good. Hence the ask.

Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/CosmosExtensions.cs]: done

**Pull request merged**